### PR TITLE
Adding celery50 testing and update import for Task

### PR DESCRIPTION
**Pull request by awais786**
Part1: https://github.com/edx/django-user-tasks/pull/106

This is Part2
https://openedx.atlassian.net/browse/BOM-2165

- Adding celery50 testing.

- `from celery.task import Task` is removed in celery5.0. Updated the `import` since its working in `4.4.7` also. Made some changes in code  [Documentation](https://docs.celeryproject.org/en/v4.4.7/internals/deprecation.html#removals-for-version-5-0)

- make upgrade using py3.8

### CODE REVIEW

File: user_tasks/signals.py [hunk 1]
```diff
@@ -75,9 +75,17 @@ def _create_chain_entry(user_id, task_id, task_class, args, kwargs, callbacks, p
             if parent_name and not parent.name:
                 parent.set_name(parent_name)
     for callback in callbacks:
-        links = callback.options.get('link', [])
-        callback_class = import_string(callback.task)
-        _create_chain_entry(user_id, callback.id, callback_class, callback.args, callback.kwargs, links, parent=parent)
+        callback_class = import_string(callback['task'])
+
+        _create_chain_entry(
```
Comment (awais786) [inline user_tasks/signals.py:80]:
```
(Pdb) callback
{'task': 'test_signals.sample_task', 'args': (1, '2'),
 'kwargs': {'user_task_name': 'Chain: 1, 2, 3'}, 
'options': {'task_id': '3fc0fef8-4090-40f6-b4df-d71e3a2add9e', 
'reply_to': 'e99174fa-afbd-379a-a239-3b67991f5115', 
'link': [test_signals.sample_task(1, '3')]}, 
'subtask_type': None, 'chord_size': None, 
'immutable': True}
```

File: user_tasks/signals.py [hunk 2]
```diff
@@ -75,9 +75,17 @@ def _create_chain_entry(user_id, task_id, task_class, args, kwargs, callbacks, p
             if parent_name and not parent.name:
                 parent.set_name(parent_name)
     for callback in callbacks:
-        links = callback.options.get('link', [])
```
Comment (awais786) [inline user_tasks/signals.py:78]:
```
(Pdb) callback
test_signals.sample_task(1, '2', user_task_name='Chain: 1, 2, 3')
(Pdb) callback.__class__
<class 'celery.canvas.Signature'>
(Pdb) callback.__dict__
{'_app': <Celery user_tasks at 0x1111d6d30>, '_type': <@task: test_signals.sample_task of user_tasks at 0x1111d6d30>, 'app': <Celery user_tasks at 0x1111d6d30>, 'type': <@task: test_signals.sample_task of user_tasks at 0x1111d6d30>, 'AsyncResult': <bound method Task.AsyncResult of <@task: test_signals.sample_task of user_tasks at 0x1111d6d30>>}
(Pdb) callback.options
{'task_id': '317f3d4b-b651-4511-9714-cd2cae325984', 'reply_to': 'ae20c3af-fd01-334c-9cc0-d47ae40325bf', 'link': [test_signals.sample_task(1, '3')]}
```

File: user_tasks/signals.py (hunk 1 above)
Comment (awais786) [reply user_tasks/signals.py:80]: I did't find any documentation for this behaviour change.

Comment (jmbowman) [reply user_tasks/signals.py:80]: The string representation may have changed, but most of the `Signature` properties should be accessible either as `dict` entries or the properties defined at https://github.com/celery/celery/blob/master/celery/canvas.py#L525-L536 .  Did you hit an error that required the change?

**Comment by awais786** [comment]
@jmbowman I got this error with master code using celery5.0
```
ERROR    celery.utils.dispatch.signal:signal.py:280 Signal handler <function create_user_task at 0x10926d670> raised: AttributeError("'dict' object has no attribute 'options'")
Traceback (most recent call last):
  File "/Users/arbisoft/.virtualenvs/django-user-taskspy38/lib/python3.8/site-packages/celery/utils/dispatch/signal.py", line 276, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/Users/arbisoft/Documents/devstack/django-user-tasks/user_tasks/signals.py", line 40, in create_user_task
    _create_chain_entry(user_id, task_id, task_class, body['args'], body['kwargs'], body['callbacks'])
  File "/Users/arbisoft/Documents/devstack/django-user-tasks/user_tasks/signals.py", line 78, in _create_chain_entry
    links = callback.options.get('link', [])
AttributeError: 'dict' object has no attribute 'options'
```

**Comment by awais786** [comment]
callback is now dict. 
```
{'task': 'test_signals.sample_task', 'args': (1, '2'), 'kwargs': {'user_task_name': 'Chain: 1, 2, 3'}, 'options': {'task_id': 'c98b9605-c509-4ebf-b8e3-eaea0c837b3c', 'reply_to': '2ed4a619-26d4-3526-8979-19c74654889b', 'link': [test_signals.sample_task(1, '3')]}, 'subtask_type': None, 'chord_size': None, 'immutable': True}

and signature class is appearing inside link 

callback['options']['link'][0].__class__<class 'celery.canvas.Signature'>

```

**Comment by jmbowman** [comment]
It looks like that was caused by https://github.com/celery/celery/pull/5954/files#diff-07a65448b2db3252a9711766beec23372715cd7597c3e309bf53859eabc0107fL416-R416 ; behavior that used to be called only in Python 2.7 environments without the `simplejson` package installed is now always used, and it replaces the original dict subclass with a plain dict after sanity checking the keys.  I think this was a mistake, those code blocks should have been removed when dropping Python 2.7 support rather than making them the default behavior.  It might be worth reporting that upstream, if you feel like taking it on.  But yes, the changes you made should handle both cases. 

That just leaves the points I noted about the `make upgrade` target changes and the changelog.

**Review by mraarif — APPROVED** [review]

**Pull request merged**